sklearn/tree/tests/test_tree.py

```diff
@@ -2225,3 +2222,93 @@ def get_joblib_pickle_non_native_endianness():
     new_clf = joblib.load(get_joblib_pickle_non_native_endianness())
     new_score = new_clf.score(X, y)
     assert np.isclose(score, new_score)
+
+
+def get_different_bitness_dtype_descr_single(dtype_descr_element):
+    dtype_name, dtype_str, *rest = dtype_descr_element
+
+    if _IS_32BIT:
+        new_dtype_str = re.sub(r"([iufc])4", r"\g<1>8", dtype_str)
+    else:
+        # Map 64 bit dtypes to their matching 32 bit counterparts:
+        # <i8 to <i4, >u8 to >u4 and so on.
+        new_dtype_str = re.sub(r"([iufc])8", r"\g<1>4", dtype_str)
+
+    return (dtype_name, new_dtype_str, *rest)
+
+
+def get_different_bitness_array(arr):
+    descr = arr.dtype.descr
```

Comment:
To follow up with informal discussions with @lesteve:  probably giving the structure of descriptors (`descr`) of  dtypes which originates from structs, e.g here:

https://github.com/scikit-learn/scikit-learn/blob/ef71c8597e3fa75340107061f65da8fe1713d6e0/sklearn/tree/_tree.pxd#L25-L34

https://github.com/scikit-learn/scikit-learn/blob/1f8825c8dd6238355191e3d1c98f4e4d54cfbf16/sklearn/tree/_tree.pyx#L66-L71

can help for maintenance?